src/main/java/dal/fatih/todoly/TaskManager.java

```diff
@@ -1,155 +1,262 @@
+                String title = resultSet.getString("title");
+                String description = resultSet.getString("description");
+                Date dueDate = resultSet.getObject("dueDate", java.sql.Date.class);
+                System.out.println("Task id : "+taskId);
+                System.out.println("Title : "+title);
+                System.out.println("Description : "+description);
+                System.out.println("Due date : "+dueDate);
+            }
+            if (counter<1) {
+                System.out.println("No task found");
+            }
+            preparedStatement.close();
+            resultSet.close();
+            connection.close();
+
+        } catch (SQLException e) {
+            System.out.println(e.getMessage());
         }
     }
 
     private void deleteTask() {
 
-        System.out.print("Task Id :");
-        String taskId = scn.nextLine();
-        Task task = tasks.get(taskId);
-        if (task != null) {
-            tasks.remove(taskId);
-            System.out.println(task.getTitle() + " titled task deleted");
-            writeTaskFile();
-        } else {
-            System.out.println("Task not found");
+        System.out.print("Task Id: ");
+        String taskIdInput = scn.nextLine();
+
+        try {
+            connection = dbConnection.getConnection();
+            String sql = "delete FROM tasks WHERE taskId=?";
+            preparedStatement = connection.prepareStatement(sql);
+            preparedStatement.setObject(1, taskIdInput);
+            int rows = preparedStatement.executeUpdate();
+
+            if (rows > 0) {
+                System.out.println("Task deleted");
+            } else {
+                System.out.println("No task found");
+            }
+            connection.close();
+            preparedStatement.close();
+        } catch (SQLDataException e) {
+            System.out.println("No task found");
+        } catch (Exception e) {
+            System.out.println(e.getMessage().substring(0, 45));
+            System.out.println(e.getStackTrace());
         }
     }
 
-    private void filterTask() {
+    private void filterTasks() {
+        System.out.print("Due Date yyyy-MM-dd (*): ");
+        String lastDateInput = scn.nextLine();
 
+        int counter = 0;
         try {
-            List<Task> foundTask = new ArrayList<>();
-            System.out.println("Last Date");
-            String lastDateInput = scn.nextLine();
-            Date lastDate = dueDateParser.parse(lastDateInput);
-
-            for (Task task : tasks.values()) {
-                if (task.getDate().before(lastDate)) {
-                    foundTask.add(task);
-                }
+            connection = dbConnection.getConnection();
+            Date date=Date.valueOf(lastDateInput);
+            String sql = "select * from TASKS where DUEDATE BETWEEN now() and ?";
+            preparedStatement = connection.prepareStatement(sql);

```

Comment:
Prepared statements should be reused. So don't keep recreating them every time you want to execute a statement. Create them for once at the beginning and keep using them.